addons/hr_expense/models/hr_expense.py

```diff
@@ -0,0 +1,317 @@
+    account_move_id = fields.Many2one('account.move', string='Ledger Posting', copy=False, track_visibility="onchange")
+    attachment_number = fields.Integer(compute='_compute_attachment_number', string='Number of Attachments')
+    reason = fields.Char(track_visibility="always")
+    state = fields.Selection([('draft', 'To Submit'),
+                              ('submit', 'Submitted'),
+                              ('approve', 'Approved'),
+                              ('post', 'Waiting Payment'),
+                              ('done', 'Paid'),
+                              ('cancel', 'Refused')
+                              ], string='Status', index=True, readonly=True, track_visibility='onchange', copy=False, default='draft', required=True,
+        help='When the expense request is created the status is \'To Submit\'.\n It is submitted by the employee and request is sent to manager, the status is \'Submitted\'.\
+        \nIf the manager approve it, the status is \'Approved\'.\n If the accountant genrate the accounting entries for the expense request, the status is \'Waiting Payment\'.')
+
+    @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
+    def _compute_amount(self):
+        for expense in self:
+            expense.untaxed_amount = expense.unit_amount * expense.quantity
+            taxes = expense.tax_ids.compute_all(expense.unit_amount, expense.currency_id, expense.quantity, expense.product_id, expense.employee_id.user_id.partner_id)
+            expense.total_amount = taxes.get('total_included')
+
+    @api.multi
+    def _compute_attachment_number(self):
+        self.attachment_number = self.env['ir.attachment'].search_count([('res_model', '=', 'hr.expense'), ('res_id', 'in', self.ids)])
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        if self.product_id:
+            description = self.product_id.name_get()[0][1]
+            if self.product_id.description:
+                description += ' '+self.product_id.description
+            self.name = description
+            self.unit_amount = self.env['product.template']._price_get(self.product_id, 'standard_price')[self.product_id.id]
+            self.product_uom_id = self.product_id.uom_id
+
+    @api.onchange('product_uom_id')
+    def _onchange_product_uom_id(self):
+        if self.product_id:
+            if self.product_uom_id.category_id != self.product_id.uom_id.category_id:
+                raise UserError(_('Selected Unit of Measure does not belong to the same category as the product Unit of Measure'))
+
+    @api.onchange('employee_id')
+    def _onchange_employee_id(self):
+        self.department_id = self.employee_id.department_id
+
+    def _add_follower(self, employee_id):
```

Comment:
can remove employee_id parameter ?
i think, we can use employee object from holiday object 

I mean
employee = self.employee_id instead of passing argument.